.github/workflows/upload-cdn.yml

```
@@ -0,0 +1,76 @@
+name: Upload packages to CDN
+
+on:
+  workflow_dispatch:
+  workflow_run:
+    workflows: ["Publish packages"]
+    types:
+      - completed
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  CDN_BUCKET: gc-design-system-production-cdn
+  CDN_REGION: ca-central-1
+
+jobs:
+  unable-to-deploy:
+      if: ${{ github.event.workflow_run.conclusion == 'failure' }}
```

Can you also check here if it's a `workflow_run` trigger just so your `workflow_dispatch` triggers still work as expected:
```suggestion
      if: ${{ github.event_name == 'workflow_run' && github.event.workflow_run.conclusion == 'failure' }}
```